test/web3/contract.js

```diff
@@ -0,0 +1,518 @@
+var json = require('./schema.js');
+
+  web3.currentProvider.enable().then(() => {
+
+  var ethblockNumber = document.getElementById("ethblockNumber");
+  var eth_gasPrice = document.getElementById("eth_gasPrice");
+  var eth_newBlockFilter = document.getElementById("eth_newBlockFilter");
+  var eth_newPendingTransactionFilter = document.getElementById("eth_newPendingTransactionFilter");
+  var eth_getUncleCountByBlockHash = document.getElementById("eth_getUncleCountByBlockHash");
+  var eth_getBlockTransactionCountByHash = document.getElementById("eth_getBlockTransactionCountByHash");
+  var eth_getTransactionCount = document.getElementById("eth_getTransactionCount");
+  var eth_getBalance = document.getElementById("eth_getBalance");
+  var eth_estimateGas = document.getElementById("eth_estimateGas");
+  var eth_getUncleCountByBlockNumber = document.getElementById("eth_getUncleCountByBlockNumber");
+  var eth_getBlockTransactionCountByNumber = document.getElementById("eth_getBlockTransactionCountByNumber");
+  var eth_uninstallFilter = document.getElementById("eth_uninstallFilter");
+  var eth_mining = document.getElementById("eth_mining");
+  var eth_submitWork = document.getElementById("eth_submitWork");
+  var eth_syncing = document.getElementById("eth_syncing");
+  var eth_getTransactionByHash = document.getElementById("eth_getTransactionByHash");
+  var eth_getTransactionByBlockHashAndIndex = document.getElementById("eth_getTransactionByBlockHashAndIndex");
+  var eth_getTransactionByBlockNumberAndIndex  = document.getElementById("eth_getTransactionByBlockNumberAndIndex");
+  var eth_getUncleByBlockHashAndIndex = document.getElementById("eth_getUncleByBlockHashAndIndex");
+  var eth_getUncleByBlockNumberAndIndex = document.getElementById("eth_getUncleByBlockNumberAndIndex");
+  var eth_getBlockByHash = document.getElementById("eth_getBlockByHash");
+  var eth_getBlockByNumber = document.getElementById("eth_getBlockByNumber");
+  
+  var eth_call = document.getElementById("eth_call");
+  var eth_getStorageAt = document.getElementById("eth_getStorageAt");
+  var eth_getCode = document.getElementById("eth_getCode")
+  var eth_getFilterChanges = document.getElementById("eth_getFilterChanges");
+  var eth_getLogs = document.getElementById("eth_getLogs");
+  var eth_accounts = document.getElementById("eth_accounts");
+  var eth_protocolVersion = document.getElementById("eth_protocolVersion");
+  var eth_sendRawTransaction = document.getElementById("eth_sendRawTransaction");
+  var eth_getTransactionReceipt = document.getElementById("eth_getTransactionReceipt");
+  
+
+  ethblockNumber.addEventListener('click', function (event) {
```

Comment:
You could add these click listeners dynamically by iterating over the methods.

One approach would be to:
- instead of exporting the object of method arrays, just export an objects that contains each of the methods
- instead of adding the html for the buttons yourself, iterate over the keys of that object and append the appropriate buttons to the dom

We can do a video call to go over this approach.

Reply:
Actually, instead of buttons, we could just make the calls and run assertions, using a test runner. Let's meet to discuss this as well.